File: Programmes/queue/Queue.java
import java.util.*;

public class Queue {
	
	public static void main(String[] args) {
		
		Que q1 = new Que(5);
		q1.enque(10);
		q1.enque(20);
		q1.enque(30);


		System.out.println(q1.deque());
		System.out.println(q1.deque());
		System.out.println(q1.deque());
		q1.enque(60);
		q1.enque(70);
		q1.enque(80);
		System.out.println(q1.deque());
		System.out.println(q1.deque());
		System.out.println(q1.deque());
		

	}
}

class Que {
	int[] que=null;
	int cap;
	int front;
	int rear;
	int size;

	public Que(int x) {
		que = new int[x];
		this.cap=x;
		this.front=-1;
		this.rear=-1;
		this.size=-1;
	}

	public void enque(int data) {
		if (size==cap-1) {
			System.out.println("QueOverFlow...");
			return;
		} else {
			rear++;
			que[rear]=data;
			size++;
			System.out.println("inserted::"+data);
		}
	}

	public int deque() {
		int item=-1;
		if (front==size) {
			System.out.println("Que UnderFlow...");
		} else {
			front++;
			item=que[front];
		}
		return item;
	}

}

File: Programmes/queue/queue.c
#include <stdio.h>
#include <stdlib.h>
#include <limits.h>
#define CAPACITY 10


//int queue[CAPACITY];
int size  = 0;
int rear  = 0; 
int front = 0;

void enque(int array[], int item) {
	if (size==CAPACITY)
	{
		printf("Overflow..\n");
	} else {
		array[rear]=item;
		rear++;
		size++;
		printf("Enqued..\n");
	}
}

int deque(int array[]) {
	int data;
	if (size==0)
	{
		printf("Underflow..\n");
	} else {
		data=array[front];
		front++;
		size--;
		printf("Dequed..\n");
		//printf("%d\n", data);
	}
	return data;
}

int main() {
	int array[CAPACITY];
	enque(array, 10);
	enque(array, 15);
	enque(array, 20);
	enque(array, 25);
	enque(array, 30);
	enque(array, 30);

	enque(array, 25);
	enque(array, 30);
	enque(array, 30);
	enque(array, 25);
	enque(array, 30);
	enque(array, 30);
	deque(array);
	deque(array);
	deque(array);
	deque(array);
	deque(array);
	deque(array);
}